tests/functional/test_network_manager.py

```diff
@@ -289,3 +289,66 @@ def test_ecosystems_include_custom(networks, custom_networks_config_dict, temp_c
         actual = networks.ecosystems
 
     assert "custom-ecosystem" in actual
+
+
+def test_fork_network_not_forkable(networks, eth_tester_provider):
+    """
+    Show correct failure when trying to fork the local network.
+    """
+    expected = "Unable to fork network 'local'."
+    with pytest.raises(NetworkError, match=expected):
+        with networks.fork():
+            pass
+
+
+def test_fork_no_providers(networks, mock_sepolia, disable_fork_providers):
+    """
+    Show correct failure when trying to fork without
+    ape-hardhat or ape-foundry installed.
+    """
+    expected = "No providers for network 'sepolia-fork'."
+    with pytest.raises(NetworkError, match=expected):
+        with networks.fork():
+            pass
+
+
+def test_fork_use_non_existing_provider(networks, mock_sepolia):
+    """
+    Show correct failure when specifying a non-existing provider.
+    """
+    expected = "No provider named 'NOT_EXISTS' in network 'sepolia-fork' in ecosystem 'ethereum'.*"
+    with pytest.raises(ProviderNotFoundError, match=expected):
+        with networks.fork(provider_name="NOT_EXISTS"):
+            pass
+
+
+def test_fork(networks, mock_sepolia, mock_fork_provider):
+    """
+    Happy-path fork test.
+    """
+    ctx = networks.fork()
+    assert ctx._disconnect_after is True
+    with ctx as provider:
+        assert provider.name == "mock"
+        assert provider.network.name == "sepolia-fork"
+
+
+def test_fork_specify_provider(networks, mock_sepolia, mock_fork_provider):
+    """
+    Happy-path fork test when specifying the provider.
+    """
+    ctx = networks.fork(provider_name="mock")
+    assert ctx._disconnect_after is True
+    with ctx as provider:
+        assert provider.name == "mock"
+        assert provider.network.name == "sepolia-fork"
+
+
+def test_fork_with_provider_settings(networks, mock_sepolia, mock_fork_provider):
```

Comment:
@mikeshultz we can test your [new feature similarly](https://github.com/ApeWorX/ape/pull/1883) similarly